sequencer/src/block2.rs

```diff
@@ -506,53 +522,46 @@ mod test {
     };
     use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
     use jf_primitives::vid::VidScheme;
+    use rand::RngCore;
 
     #[test]
     fn basic_correctness() {
         // play with this
         let test_cases = vec![
-            // 3 non-empty txs
-            vec![
-                vec![0, 1, 2, 3, 4],
-                vec![5, 6, 7, 8, 9, 10, 11, 12],
-                vec![13, 14, 15, 16, 17, 18, 19, 20],
-            ],
-            // 1 empty tx at the beginning
-            vec![
-                vec![],
-                vec![5, 6, 7, 8, 9, 10, 11, 12],
-                vec![13, 14, 15, 16, 17, 18, 19, 20],
-            ],
-            // 1 empty tx in the middle
-            vec![
-                vec![0, 1, 2, 3, 4],
-                vec![],
-                vec![13, 14, 15, 16, 17, 18, 19, 20],
-            ],
-            // 1 empty tx at the end
-            vec![vec![0, 1, 2, 3, 4], vec![5, 6, 7, 8, 9, 10, 11, 12], vec![]],
-            // 1 nonempty tx
-            vec![vec![0, 1, 2, 3, 4]],
-            // 1 empty tx
-            vec![vec![]],
-            // zero txs
-            vec![],
+            vec![5, 13, 21],        // 3 non-empty txs
+            vec![0, 8, 16],         // 1 empty tx at the beginning
+            vec![5, 5, 13],         // 1 empty tx in the middle
```

Comment:
```suggestion
            vec![5, 0, 13],         // 1 empty tx in the middle
```
?

Reply:
These vecs are verbatim tx table entries. So `5,5` means that the tx starts and ends at index 5. I did it this way to allow for negative tx lengths in the tx table. But perhaps it is a bit confusing. Maybe it's better to have 2 different helpers: one that takes *lengths* as input and another that takes *table entries* as input.

Reply:
added `entries_from_lengths` in https://github.com/EspressoSystems/espresso-sequencer/pull/846/commits/ab2a58b8e7bc9ac779722b3b28f3c108ac6b60b5

Reply:
Oh yeah I figured this out later in my reading and forgot to come back to this comment